### PR TITLE
Allow to access passProps in registerCommandListener

### PR DESCRIPTION
**Pull request by jfrolich**
see issue: https://github.com/wix/react-native-navigation/issues/4716

### CODE REVIEW

File: lib/src/commands/Commands.ts [hunk 1]
```diff
@@ -86,11 +87,12 @@ export class Commands {
     const input = _.cloneDeep(simpleApi);
 
     const layout = this.layoutTreeParser.parse(input);
+    const copiedLayout = _.cloneDeep(layout);
     this.layoutTreeCrawler.crawl(layout);
 
     const commandId = this.uniqueIdProvider.generate('push');
     const result = this.nativeCommandsSender.push(commandId, componentId, layout);
-    this.commandsObserver.notify('push', { commandId, componentId, layout });
+    this.commandsObserver.notify('push', { commandId, componentId, copiedLayout });
```
Comment (guyca) [inline lib/src/commands/Commands.ts]: @jfrolich This line broke the tests, it should probably be `layout: copiedLayout`

**Comment by jfrolich** [comment]
Yes. Sorry had to retype the commit after prettier changed all the formatting, introducing this typo. Should be good now.

**Comment by guyca** [comment]
Hey @jfrolich 
I just realised that this change is a bit problematic. Your commit changes things so that the original layout (which includes passProps) is passed to the command - but since it hasn't been processed by RNN, **layout nodes don't include id's**.

We omit passProps because we don't want to pass them over the bridge. As they might be very big (think about passing base64 image in passProps), they might put unneeded strain on the bridge.

We can't merge your solution as it currently is. What we can do, is add a function to `LayoutTreeCrawler` which will add props to components after the command has passed over the bridge, and before we notify observers.
WDYT?

**Comment by jfrolich** [comment]
Hmm yeah I think we need to seperate removing the passProps from layoutCrawler, so we can deepClone in between.

**Comment by guyca** [comment]
Alright, I'm leaving the PR open in case you'll want to reiterate. I'm also available on discord if you want to consult with me while working on the PR 👍

**Comment by jfrolich** [comment]
Yes, will do

**Comment by jfrolich** [comment]
Need to keep this one active.

**Comment by guyca** [comment]
This was fixed in a different PR and already available in master 👍

**Pull request closed without merging**